src/py_gc.rs

```diff
@@ -0,0 +1,70 @@
+use ahash::AHashMap;
+use enum_dispatch::enum_dispatch;
+use pyo3::{AsPyPointer, Py, PyTraverseError, PyVisit};
+
+/// Trait implemented by types which can be traversed by the Python GC.
+#[enum_dispatch]
+pub trait PyGcTraverse {
+    fn py_gc_traverse(&self, visit: &PyVisit<'_>) -> Result<(), PyTraverseError>;
+}
+
+impl<T> PyGcTraverse for Py<T>
+where
+    Py<T>: AsPyPointer,
+{
+    fn py_gc_traverse(&self, visit: &PyVisit<'_>) -> Result<(), PyTraverseError> {
+        visit.call(self)
+    }
+}
+
+impl<T: PyGcTraverse> PyGcTraverse for Vec<T> {
+    fn py_gc_traverse(&self, visit: &PyVisit<'_>) -> Result<(), PyTraverseError> {
+        for item in self {
+            item.py_gc_traverse(visit)?;
+        }
+        Ok(())
+    }
+}
+
+impl<T: PyGcTraverse> PyGcTraverse for AHashMap<String, T> {
+    fn py_gc_traverse(&self, visit: &PyVisit<'_>) -> Result<(), PyTraverseError> {
+        for item in self.values() {
+            item.py_gc_traverse(visit)?;
+        }
+        Ok(())
+    }
+}
```

Comment:
Seems like these might even make sense in PyO3 at some point, right?

Reply:
Yes, though probably not in this form. In PyO3 I'd like (if it's possible) to do it automatically with a `#[derive]` macro, however the challenge is how to make most types noops and enable gc traversal for just the ones that matter. I'm sure there's an issue about this in the PyO3 backlog but I can't find it right now.